src/main/webapp/3Dscript.js

```diff
@@ -0,0 +1,183 @@
+ */
+function initThreeJs() {
+  scene = new THREE.Scene();
+  camera = new THREE.PerspectiveCamera(
+      /* fov =*/ 75,
+      /* aspectRatio =*/ window.innerWidth / window.innerHeight,
+      /* nearFrustum =*/.1,
+      /* farFrustum =*/ 1000);
+  camera.position.set(0, 1, 1);
+  renderer = new THREE.WebGLRenderer();
+  renderer.setSize(window.innerWidth, window.innerHeight);
+  document.body.appendChild(renderer.domElement);
+  clock = new THREE.Clock();
+  controls = new OrbitControls(camera, renderer.domElement );
+
+  fetchData();
+}
+
+
+/**
+ * This function fetchs pose data from the RunInfo servlet,
+ * it is an asynchronous call requiring addTerrain() to be 
+ * called after the data is fully loaded.
+ */
+function fetchData() {
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const id = urlParams.get('id');
+    const type = urlParams.get('dataType');
+    fetch('/getrun?id=' + id + '&dataType=' + type)
+    .then(response => response.json())
+    .then(data => pose = data)
+    .then(() => addTerrian());
+}
+
+/**
+ * Creates the 2D terrain and points a light at it.
+ * It also calls two other functions so it can wait on the fetch 
+ * request and significantly cut down rendering time with 
+ * geometry instancing.
+*/
+function addTerrian() {
+  // Creates the 2D terrain.
+  dataLength = pose.length;
+  let loader = new THREE.TextureLoader();
+  let material = new THREE.MeshLambertMaterial({ 
+    map: loader.load(
+      'https://maps.googleapis.com/maps/api/staticmap' +
+      '?format=png&center=48.129872,11.582905&zoom=18&size=500x500&key='
+      + apiKey)
+    });
+  let geometry = new THREE.PlaneGeometry(50,50);
+  const map = new THREE.Mesh(geometry, material);
+  map.position.set(0,0,0);
+  map.rotation.x = -Math.PI / 2;
+  scene.add(map);
+
+  // Add the light to the scene.
+  let light = new THREE.PointLight(0xffffff, 1, 0 );
+  light.position.set(0, 100, 0);
+  scene.add(light)
+  console.log(dataLength);
+  plotTrajectory();
+  
+  // Adds pose data to the scene as one Geometry instance.
+  material = new THREE.MeshBasicMaterial({color: 'red'});
+  geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
+  orientation = new THREE.InstancedMesh(geometry, material, dataLength);
+  orientation.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation);
+  plotOrientation();
+}
+
+/**
+ * This uses the pose data to create a blue line representing the pose
+ * trajectory. 
+ */
+function plotTrajectory() {
+  // Removes any existing trajectory objects for repositioning.
+  scene.remove(trajectory);
+  let coordinates=[];
+  for (let increment = 0; increment < dataLength; increment++) {
+    coordinates.push(new THREE.Vector3(
+      (pose[increment].lng-11.582905) * 25000 * poseScalar.multiplier,
+      (pose[increment].alt - 6.582905)/4,
+      (pose[increment].lat-48.129872) * 25000 * -poseScalar.multiplier));// GPS points
```

Comment:
What are these magic offsets and multipliers?

Reply:
would these be the numbers that are just place holders for until an appropiate conversion of ecef can be found?

Reply:
The offset is just the first value in the pose data, Ill replace that with pose[0].lat/lng/alt. The diameter of our 2D map in the real world is 200 meters while the 2D plane in our scene is 50 units, therefore each meter is about 1/4th of a unit. 25000 is the constant to scale the data up to meters, given the 6th decimal in GPS coordinates represents about .11 meters. Like Frelicia mentioned, I'm hoping to add a better unit and conversion system fairly soon.

Reply:
Thanks. Definitely prioritize a more accurate conversion, since degrees of latitude and longitude don't represent consistent distances across the earth. For this PR, the approximation is fine.